### ISSUE TITLE
AimLogger `save_dir` AttributeError in PyTorch Lightning Parallel GPU Training

### ISSUE THREAD

**Issue by GeeeekExplorer** [aimhubio/aim#1378]
## 🐛 Bug

I use `AimLogger` in PL's `Trainer` with options `gpus=2` and `strategy='ddp'`. Here is brief traceback:

 Traceback (most recent call last):                                                                                          
  ......
  File "....../lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py", line 1572, in __setup_profiler
    self.profiler.setup(stage=self.state.fn._setup_fn, local_rank=local_rank, log_dir=self.log_dir)
  File "....../lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py", line 1768, in log_dir
    dirpath = self.logger.save_dir
  File "....../lib/python3.9/site-packages/aim/sdk/adapters/pytorch_lightning.py", line 107, in save_dir
    return self.experiment.repo.path
AttributeError: 'function' object has no attribute 'path‘

### To reproduce

It is easy to reproduce. Just modify line **100** of [examples/pytorch_lightning_track.py](https://github.com/aimhubio/aim/blob/8d91981d866087e727707bdb204c412be2b38c97/examples/pytorch_lightning_track.py#L100):
from
```python
trainer = pl.Trainer(logger=aim_logger)
```
to
```python
trainer = pl.Trainer(gpus=2, strategy='ddp', logger=aim_logger)
```

### Expected behavior

Parallel training works well.

### Solution

Brief bug-related code in [aim/sdk/adapters/pytorch_lightning.py](https://github.com/aimhubio/aim/blob/8d91981d866087e727707bdb204c412be2b38c97/aim/sdk/adapters/pytorch_lightning.py#L20):
```python
class AimLogger(LightningLoggerBase):
    @property
    def save_dir(self) -> str:
        return self.experiment.repo.path

    @property
    @rank_zero_experiment
    def experiment(self) -> Run:
        if self._run is None:
            self._run = Run(
                repo=self._repo_path,
                experiment=self._experiment_name,
                system_tracking_interval=self._system_tracking_interval
            )
        return self._run
```

Definition of decorator `rank_zero_experiment` in PyTorch Lightning:
```python
def rank_zero_experiment(fn: Callable) -> Callable:
    """Returns the real experiment on rank 0 and otherwise the DummyExperiment."""

    @wraps(fn)
    def experiment(self):
        @rank_zero_only
        def get_experiment():
            return fn(self)

        return get_experiment() or DummyExperiment()

    return experiment
```

So in a process which is not rank 0, `self.experiment` is a `DummyExperiment` object, which leads to bug.

The bug can be fixed by explicitly computing path as [aim/sdk/run.py](https://github.com/aimhubio/aim/blob/8d91981d866087e727707bdb204c412be2b38c97/aim/sdk/run.py#L764):
```python
    @property
    def save_dir(self) -> str:
        from aim.sdk.repo import Repo
        return self._repo_path or Repo.default_repo_path()
```

### Environment

- Aim 3.5.4
- Python 3.9.7
- pip 22.0.3
- Ubuntu 20.04

**Comment by mihran113** [comment]
hey @GeeeekExplorer. Thanks for the report and detailed steps to reproduce and fix! We'll include the fix with in upcoming patch release `v3.6.1` or `v3.6.2`.

**Comment by mihran113** [comment]
hey @GeeeekExplorer. Thanks a lot for your contribution. Just wanted to inform you that the fix you've provided was shipped in `v3.6.1`.
